receiver/snmpreceiver/config.go

```diff
@@ -0,0 +1,592 @@
+				combinedErr = multierr.Append(combinedErr, validateColumnOID(metricName, columnOID, cfg))
+			}
+		}
+	}
+
+	return combinedErr
+}
+
+// validateColumnOID validates a ColumnOID
+func validateColumnOID(metricName string, columnOID ColumnOID, cfg *Config) error {
+	var combinedErr error
+
+	// Ensure that it contains an OID
+	if columnOID.OID == "" {
+		combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnOIDNoOID, metricName))
+	}
+
+	// Keep track of whether the different indexed values can be differentiated by either attribute within the same metric
+	// or by different resource attributes (in different resources)
+	hasIndexedIdentifier := false
+
+	// Check that any Attributes have a valid Name and a valid Value (if applicable)
+	if len(columnOID.Attributes) > 0 {
+		for _, attribute := range columnOID.Attributes {
+			if attribute.Name == "" {
+				combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnAttributeNoName, metricName))
+			} else {
+				attrCfg, ok := cfg.Attributes[attribute.Name]
+				if !ok {
+					combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnAttributeBadName, metricName, attribute.Name))
+				} else {
+					if len(attrCfg.Enum) > 0 {
+						if !contains(attrCfg.Enum, attribute.Value) {
+							combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnAttributeBadValue, metricName, attribute.Name, attribute.Value))
+						}
+					} else {
+						hasIndexedIdentifier = true
+					}
+					if len(attrCfg.Enum) != 0 && !contains(attrCfg.Enum, attribute.Value) {
+						combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnAttributeBadValue, metricName, attribute.Name, attribute.Value))
+					}
+				}
+			}
+		}
+	}
+
+	// Check that any ResourceAttributes have a valid value
+	if len(columnOID.ResourceAttributes) > 0 {
+		hasIndexedIdentifier = true
+		for _, name := range columnOID.ResourceAttributes {
+			_, ok := cfg.ResourceAttributes[name]
+			if !ok {
+				combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnResourceAttributeBadName, metricName, name))
+			}
+		}
+	}
+
+	if !hasIndexedIdentifier {
+		// TODO: correct error and write test
+		combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnIndexedAttributeRequired, metricName))
+	}
+
+	return combinedErr
+}
+
+// validateScalarOID validates a ScalarOID
+func validateScalarOID(metricName string, scalarOID ScalarOID, cfg *Config) error {
+	var combinedErr error
+
+	// Ensure that it contains an OID
+	if scalarOID.OID == "" {
+		combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgScalarOIDNoOID, metricName))
+	}
+
+	if len(scalarOID.Attributes) == 0 {
+		return combinedErr
+	}
+
+	// Check that any Attributes have a valid Name and a valid Value (if applicable)
+	for _, attribute := range scalarOID.Attributes {
+		if attribute.Name == "" {
+			combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgScalarAttributeNoName, metricName))
```

Comment:
Same thing here as if we can return early we can clean up some nesting.

Reply:
flattened conditionals